src/tree-sitter/ExpressionBuilder.ts

```diff
@@ -130,12 +131,25 @@ function toString(s: string): string {
 function recordFromMatch<T extends string>(
   match: Parser.QueryMatch,
   keys: readonly T[]
-): Record<T, string | undefined> {
-  const values = keys.map((name) => match.captures.find((c) => c.name === name)?.node?.text)
-  return Object.fromEntries(keys.map((_, i) => [keys[i], values[i]])) as Record<
+): RecordLink {
+  const recordValues = keys.map((name) => match.captures.find((c) => c.name === name)?.node?.text)
+  const record = Object.fromEntries(keys.map((_, i) => [keys[i], recordValues[i]])) as Record<
     T,
     string | undefined
   >
+
+  // const linkValues = match.captures
```

Comment:
I need some explanation how this works and how I can be sure to always match the right step definitions (also the ones with expression.

Perhaps you can right the test with proper data and I will finish the code?

Reply:
Yes, I'll have a look into this.